go/vt/discovery/healthcheck.go

```diff
@@ -521,6 +521,7 @@ func (hcc *healthCheckConn) stream(ctx context.Context, hc *HealthCheckImpl, cal
 		if hc.listener != nil {
 			hc.listener.StatsUpdate(&ts)
 		}
+		conn.Close(ctx)
```

Comment:
I think this should be:
within lock:
```
hccConn := hcc.conn
hcc.conn = nil
```

and `hccConn.Close(ctx)` here.

Reply:
I changed this as you suggested, but I can't think of any case in which `conn` != `hcc.conn` here and IMO this actually detracts from readability.

Reply:
For the record I removed this change again because  we discussed in Slack and agreed that the original version was just as safe and was cleaner.